app/src/main/java/org/simple/clinic/drugs/selectionv2/entry/CustomPrescriptionEntryControllerv2.kt

```diff
@@ -173,4 +176,32 @@ class CustomPrescriptionEntryControllerv2 @Inject constructor(
           }
         }
   }
+
+  private fun removePrescription(events: Observable<UiEvent>): Observable<UiChange> {
+    val openAsUpdate = events
+        .ofType<CustomPrescriptionSheetCreated>()
+        .filter { it.openAs is OpenAs.Update }
+        .map { it.openAs as OpenAs.Update }
+        .map { it.prescribedDrugUuid }
+        .take(1)
+
+    return events
+        .ofType<RemoveCustomPrescriptionClicked>()
+        .withLatestFrom(openAsUpdate)
+        .map { (_, prescribedDrugUuid) -> { ui: Ui -> ui.showConfirmRemoveMedicineDialog(prescribedDrugUuid) } }
+  }
+
+  private fun closeSheetWhenPrescriptionIsDeleted(events: Observable<UiEvent>): Observable<UiChange> {
+    val prescribedDrugUuuids = events
+        .ofType<CustomPrescriptionSheetCreated>()
+        .filter { it.openAs is OpenAs.Update }
+        .map { it.openAs as OpenAs.Update }
+        .map { it.prescribedDrugUuid }
+
+    return prescribedDrugUuuids
+        .flatMap(prescriptionRepository::prescription)
+        .filter { it.isDeleted }
```

Comment:
Nice! This sheet gets finished as soon as the observed item is removed. Persistence makes state management so easy!